ctera/src/main/java/ch/cyberduck/core/ctera/CteraDeleteFeature.java

```diff
@@ -0,0 +1,36 @@
+package ch.cyberduck.core.ctera;
+
+/*
+ * Copyright (c) 2002-2024 iterate GmbH. All rights reserved.
+ * https://cyberduck.io/
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+import ch.cyberduck.core.Path;
+import ch.cyberduck.core.dav.DAVDeleteFeature;
+import ch.cyberduck.core.exception.BackgroundException;
+
+import static ch.cyberduck.core.ctera.CteraAclPermissionFeature.DELETEPERMISSION;
+import static ch.cyberduck.core.ctera.CteraAclPermissionFeature.checkCteraRole;
+
+public class CteraDeleteFeature extends DAVDeleteFeature {
+    public CteraDeleteFeature(final CteraSession session) {
+        super(session);
+    }
+
+    @Override
+    public void preflight(Path file) throws BackgroundException {
```

Comment:
@dkocher do we not need integration tests for all the overriden `preflight`s?

Reply:
We need, currently waiting for test environment with proper support. Currently we can test that no preflight check is throwing.